assets/js/modules/pagespeed-insights/components/dashboard/DashboardPageSpeed.js

```diff
@@ -232,126 +290,171 @@ export default function DashboardPageSpeed() {
-					<FieldReportMetrics
-						data={ reportData }
-						error={ reportError }
-					/>
+								<span
+									id={ `googlesitekit-pagespeed-widget__data-src-tab-${ DATA_SRC_FIELD }` }
+									className="mdc-tab__text-label"
+								>
+									{ __( 'In the Field', 'google-site-kit' ) }
+								</span>
+							</Tab>
+							<Tab
+								focusOnActivate={ false }
+								aria-labelledby={ `googlesitekit-pagespeed-widget__data-src-tab-${ DATA_SRC_RECOMMENDATIONS }` }
+							>
+								<span
+									id={ `googlesitekit-pagespeed-widget__data-src-tab-${ DATA_SRC_RECOMMENDATIONS }` }
+									className="mdc-tab__text-label"
+								>
+									{ __(
+										'How to improve',
+										'google-site-kit'
+									) }
+								</span>
+							</Tab>
+						</TabBar>
+					</div>
+					<div className="googlesitekit-pagespeed-widget__device-size-tab-bar-wrapper">
+						<DeviceSizeTabBar
+							activeTab={ strategy }
+							handleDeviceSizeUpdate={ updateActiveDeviceSize }
+						/>
+					</div>
+				</header>
+				{ isFetching && ! isLoading && (
+					<div className="googlesitekit-pagespeed-widget__refreshing-progress-bar-wrapper">
+						<ProgressBar compress />
+					</div>
 				) }
-			</section>
 
-			{ ! reportError && (
-				<Recommendations
+				<section
 					className={ classnames( {
 						'googlesitekit-pagespeed-widget__refreshing':
 							isFetching,
 					} ) }
-					referenceURL={ referenceURL }
-					strategy={ strategy }
-				/>
-			) }
+				>
+					{ dataSrc === DATA_SRC_LAB && (
+						<LabReportMetrics
+							data={ reportData }
+							error={ reportError }
+						/>
+					) }
+					{ dataSrc === DATA_SRC_FIELD && (
+						<FieldReportMetrics
+							data={ reportData }
+							error={ reportError }
+						/>
+					) }
+					{ dataSrc === DATA_SRC_RECOMMENDATIONS && (
+						<Recommendations
+							className={ classnames( {
+								'googlesitekit-pagespeed-widget__refreshing':
+									isFetching,
+							} ) }
+							recommendations={ recommendations }
+							referenceURL={ referenceURL }
+							strategy={ strategy }
+						/>
+					) }
+				</section>
 
-			<div
-				className={ classnames(
-					'googlesitekit-pagespeed-report__footer',
-					{
-						'googlesitekit-pagespeed-report__footer--with-action':
-							dataSrc === DATA_SRC_LAB,
-					}
-				) }
-			>
-				{ dataSrc === DATA_SRC_LAB && (
-					<div>
-						<Link onClick={ updateReport } disabled={ isFetching }>
-							{ __( 'Run test again', 'google-site-kit' ) }
-						</Link>
-						<Spinner isSaving={ isFetching } />
+				{ ( dataSrc === DATA_SRC_LAB ||
+					dataSrc === DATA_SRC_FIELD ) && (
+					<div className="googlesitekit-pagespeed-report__row">
+						<Button
+							className={ classnames( {
+								'googlesitekit-pagespeed__recommendations-cta--hidden':
+									! recommendations?.length,
+							} ) }
+							onClick={ () =>
+								updateActiveTab( TAB_INDEX_RECOMMENDATIONS )
+							}
```

Comment:
Actually in this particular case there's no need for a `useCallback`. The `Button` component is not memoized and so it will always rerender regardless of whether it's passed the same or a new function instance here.

As `updateActiveTab` is passed directly elsewhere its `useCallback` is still just as relevant as it previously was. But there's no point introducing another `useCallback`, it would add complexity and a little overhead but serve no purpose.

As it happens, I saw your comment last night and wanted to refresh my memory on this sort of thing. I found and read [this article](https://www.developerway.com/posts/how-to-use-memo-use-callback) which was a useful refresher. In general I think we are probably using `useCallback` a bit excessively and should be a bit more considered about where we choose to use it...

Reply:
Yeah, we have this discussion a year ago but then kept using useCallbacks. Ok, let it be it. Sounds good to me.